### PR TITLE
fix(frontend): router redirects

### PR DESCRIPTION
**Pull request by spion**
This fixes router redirects breaking the back button. Browser history is now also set to memory history in testing mode, which enables full testing of the router by driving the history instead of the route provider.

The router was modified to use computed values instead of reactions to determine the components, and use autorun only for side effects that dispatch changes to external modules (i.e. history)

### CODE REVIEW

File: packages/frontend/src/router/router.tsx [hunk 1]
```diff
@@ -101,27 +89,26 @@ class MobxRouter {
       return params;
     };
     this.routes.unshift({ match, component });
-  };
+  }
 
-  @action
-  addRedirect = (newRedirect: Redirect) => {
+  @action.bound
```
Comment (EmilH4) [inline packages/frontend/src/router/router.tsx]: Why is this changed?

Comment (spion) [reply packages/frontend/src/router/router.tsx]: A very silly reason: since its a JSX file, TypeScript was trying to interpret my `<T>` argument as the beginning of a JSX block when I wrote `addRedirect = <T>(newRedirect:Redirect)`

**Comment by spion** [comment]
Yeah it appears a reference to the test renderer isn't actually necessary - I removed withTestRenderer.
